# Models of Binary Dependent Variable 
Lecture notes by
_Sunil Paul_

- The materials in this notebook is prepared using Scott Long's "Regression models for categorical and Limited dependent variables".The figure posted in the notes are sourced from this book.
- Kindly refrain from circulating this note outside the class, as I haven't included citations for all the textbooks/paper I referenced in the preparation of these notes.
- Also see Diebold, F.X. (2019), Econometric Data Science: A Predictive Modeling Approach, Department of Economics, University of Pennsylvania, http://www.ssc.upenn.edu/~fdiebold/Textbooks.html.”-


Binary dependent variable models are utilized when outcomes or responses are discrete, typically taking values of 1 or 0. Examples include decisions such as whether to take a job or not, whether to pursue higher studies, and similar scenarios. These models are particularly suited for situations where the outcome can be categorized into two distinct groups.

## The Linear Probability Model (LPM) 
When employing Ordinary Least Squares (OLS) to estimate parameters in the context of a binary or dummy variable as the dependent variable, the resultant regression is referred to as the Linear Probability Model (LPM). In the LPM, the dependent variable is binary, signifying outcomes such as the presence or absence of a particular event. For instance, this could encompass scenarios where the event of interest either happens or does not occur. 

It can be expressed as $$y=X\beta +\varepsilon,$$ where $E(y|X)=X\beta$. 
For instance, suppose we aim to investigate the determinants of whether a woman is engaged in paid employment or not. In this case, the Linear Probability Model (LPM) can be specifically formulated as follows:  $$y_i=\beta_{0}+\beta_{1}x_{1i}+...+\beta_{k}x_{ki},$$ where, $\beta$s are the parameters of interest and $x_i$s are the independent variables and  $y_{i}$ is defined as:$$
y_{i}=\left\{ 
\begin{array}{l}
1-\text{if  ith woman is in paid employment} \\ 
0-\text{otherwise}
\end{array}
\right. $$

### Why it is linear probability model?
To comprehend why this model is termed the Linear Probability Model (LPM), it's essential to delve into the concepts of unconditional and conditional expectations of the dependent variable $y_{i}$. 
The unconditional expectation of $y_{i}$, denoted as $E(y_{i})$, is computed by summing the probabilities of $y_{i}$ taking the value 1 or 0, weighted by their respective occurrences. Mathematically, it is expressed as:


$$E(y_{i})=[1\times P(y_{i}=1)]+[0\times P(y_{i}=0)]=P(y_{i}=1).$$ Similarly, the conditional expectation of $y_{i}$ given $x_{i}$, denoted as $E(y_{i}|x_{i})$, involves calculating the probabilities of $y_{i}$ being 1 or 0 given the values of $x_{i}$, and summing them up. $$E(y_{i}|x_{i})=[1\times P(y_{i}=1|x_{i})]+[0\times P(y_{i}=0|x_{i})]=P(y_{i}=1|x_{i})$$. Hence, $E(y_{i}|x_{i})=\alpha +\beta x_{i}=P(y_{i}=1|x_{i})=P_{i}$, where $\beta$ represents the expected change in the probability of being in paid employment due to a one-unit change in $x_{i}$.

In essence, $E(y_{i}|x_{i})$ provides the probability of the occurrence of the event of interest, given the specific conditions or characteristics denoted by $x_{i}$. In the context of analyzing whether a woman is engaged in paid employment, $E(y_{i}|x_{i})$ signifies the likelihood of being in paid employment based on the particular attributes represented by $x_{i}$. Therefore, the name LPM

In matrix notation, the Linear Probability Model (LPM) can be represented as $E(y|X)=X\beta =P$, where $y$ is a vector containing $n$ observations, and $X$ is the matrix of independent variables as defined in the previously circulated notes. Each element $\beta_{k}$ in the coefficient vector $\beta$ quantifies the expected change in the probability of the event occurring due to a one-unit change in the corresponding independent variable $x_{k}$, with all other variables held constant.

A notable characteristic of the LPM is that a unit change in any independent variable $x_{k}$ consistently leads to the same change in the probability, regardless of the specific variable being altered. This uniform relationship between changes in the independent variables and alterations in the probability of the event occurring gives rise to the term 'Linear Probability Model'

### Limitations of LPM
However, the LPM has its limitations. One significant issue pertains to the properties of errors. The errors $\varepsilon _{i}$ can take two values based on the outcome $y_{i}$, leading to non-normal distributions and heteroskedasticity. i.e:
$$\varepsilon _{i}=\left\{ 
\begin{array}{l}
1-X\beta \text{ if }y_{i}=1\text{ with probability }P_{i} \\ 
-X\beta \text{ if }y_{i}=0\text{ with probability }1-P_{i}
\end{array}
\right. $$
and the expection and variance of errors  can be derived as follows:
$$E(\varepsilon|X)=P(1-X\beta )+(1-P)(-X\beta
) ,$$ substituting  $P=X\beta $ we get

$$E(\varepsilon|X)=0,$$
and
$$\begin{align*}
\text{var}(\varepsilon|X) &= P(1-X\beta)^{2} + (1-P)(-X\beta)^{2} \\
&= X\beta (1-X\beta)^{2} + (1-X\beta)(-X\beta)^{2} \\
&= X\beta (1-X\beta)
\end{align*}.$$
Since $\text{var}(\varepsilon|X) $ depends on X, its heteroschedastic. Hence, std. errors are biased and OLS is inefficient.

Additionally, the interpretation of coefficients in the LPM may not always align with expectations, as the assumption of linearity in probability can be unrealistic in certain contexts. Also, $E(y|X)=X\beta $ is interpreted as probabilities and in many cases $X\beta $ lie outside the interval (0,1).

#  Latent variable intrepretation 

In the scenario outlined above, consider an individual's labour force participation as the observed $y$. The variable $y$ can only be observed in two states: an individual is in the labour force or she is not. However, not all  in the labor force are there with the same certainty. One individual may be very close to the decision of leaving the labour force, while another individual could be very firm on her decision. In both cases, we observe the same $y=1$. The idea of a latent $y^*$ is that there is an underlying propensity to work that generates the observed state. While we cannot directly observe $y^*$ , at some point a change in $y^*$  results in a change in what we observe, namely whether an individual is in the labor force. To clarify, let's assume that individual $i$ has an expected wage (reservation wage), and they decide to enter employment if the market wage exceeds this expectation. Let $w_i$ represent the market wage, and $w_i^*$. Then the latent variable $y_i^*$ can be expressed as follows: 
$$y_{i}^{\ast }=w_{i}- w_{i}^{\ast }$$
Given this,
$$y_{i}=\left\{ 
\begin{array}{l}
1\text{  if }y_{i}^{\ast }>0 \\ 
0\text{ if }y_{i}^{\ast }\leq 0
\end{array}
\right. $$

The latent variable $y^*$ is assumed to be linearly related to observed $x$s through the model as follows:
$$y^*=X\beta+\varepsilon$$

In general; $y_{i}^{\ast }=X_{i}\beta +\varepsilon _{i}$ and

$$y_{i}=\left\{ 
\begin{array}{l}
1\ \text{if }y_{i}^{\ast }>\tau \\ 
0\ \text{if }y_{i}^{\ast }\leq \tau
\end{array}
\right. $$



where$tau$ is the threshold or cutpoint. If $y^* \le \tau$, then $y=0$. If $y^*$ crosses the threshold $\tau$(i.e., $y^*> \tau$) then $y=1$.(We assume $\tau=0$ for simplicity)

![Example Image](fig1.png)

The link between the latent $y^*$ is illustrated in the figure above for the model $y^*=\beta_0+\beta_1x+\varepsilon$. $y^*$ is on the vertical axis, with the threshold $\tau$ indicated by a horizontal dashed line. The distribution of $y^*$ shown by a bell shaped curve. When $y^*$ is larger than $\tau$ indicated by the shaded region, we observe $y=1$.

We cannot use Ordinary Least Squares (OLS) regression because the dependent variable is unobserved. Instead, we must use Maximum Likelihood Estimation (MLE), which requires specific assumptions about the errors. For example, in the Probit model, we assume the errors are normally distributed, while in the Logit model, we assume the errors follow a logistic distribution.

## Probit  and Logit Model
Distributional assumptions regarding $\varepsilon$. However since $y^*$ is unobserved, we cannot estimate the variance of the errors. Instead, we need to assume some values for the variance of errors. We assume the $E(\varepsilon|X)=0$ and if  we assume $var(\varepsilon|X)=1$ it is a probit model and if we assume $var(\varepsilon|X)=\frac{\pi^2}{3}\approx3.29$ then it is a probit model. Probit model assumes standrard normal distribution for the errors and the pdf and cdf are given as follows:
The pdf of errors in a probit model is:$$\phi (\varepsilon )=\frac{1}{\sqrt{2\pi }}\exp
\left( -\frac{\varepsilon ^{2}}{2}\right) $$

and the cdf :$$\Phi (\varepsilon )=\int\limits_{-\infty }^{\varepsilon }%
\frac{1}{\sqrt{2\pi }}\exp \left( \frac{z_{i}^{2}}{2}\right) dz_{i}.$$



In the logit model errors are logistically distributed. The pdf of the error distribution is given below:
$$\lambda (\varepsilon )=\frac{\exp (\varepsilon )}{[1+\exp
(\varepsilon )]^{2}}$$

and the cdf:$$\Lambda (\varepsilon ) =\frac{\exp (\varepsilon )}{1+\exp (\varepsilon )}$$

Under the distributional assumptions, we posit that the latent variable$y^*$  follows a normal or logistic distribution, while its realization $y$ does not. However, we can infer the probability $P(y|X=1)$ using the error distribution. Specifically, when $y^*>0$, the probability $P(y|X=1)$ can be derived from the known probability density function (pdf) of $y^*>0$ determined by the error distribution. This is illustrated in the figure below. Given $\varepsilon$ is distributed either logistically or normaaly around $E(y^*|x)=\beta_0+\beta_1x$. Values of $y=1$ are observed for the shaded portion of error distribution above $\tau$

![Example Image](fig2.png)

The error distribution given in the figure above can be presented below to clarify the link. Since $y=1$ when $y^*>0,$ (Panel A in the figure)  $${P}(y=1|X)=P(y^{\ast }>0|X)$$

![Example Image](fig3.png)

Substituting $y^{\ast} = X\beta + \varepsilon$
 if follows that
$${P}(y=1|X)=P(X\beta+\varepsilon>0|X)$$
Substracting $X\beta$ from each side of the inequality shifts the x-axis as shown in Panel B. Then $${P}(y=1|X)=P(\varepsilon >-X\beta |X)$$
 Since cdf's express the probability of a variable being less than some value we must change the direction of the inequality. The normal and logistic distributions are symmetric, which means that the shaded area of the distribution greater than $-X\beta$ in Panel B equals the shaded area less than $X\beta$ in panel C. Consequently.
 $${P}(y=1|X)=P(\varepsilon \le X\beta |X)$$

This is simply the cdf of the error distribution evaluated at $X\beta$. Accordingly, $$P(y=1|X)=F(X\beta)$$

As each $x_i$ in $X$ increases by one unit, the argument of $F$ increases by $\beta_i$ units, and $F(\cdot) \in [0,1]$. The plot below illustrates the probit/logit regression. Panel A shows the error distribution of nine values of $x$. The region of the distribution where $y^* > \tau$ corresponds to $P(y=1|x)$. At $x_1$, only a small portion of the tail of the curve crosses the threshold in panel A, resulting in a small value for $P(y=1|x)$ in the panel B below. As we move to $x_2$, the error distribution shifts up slightly.

![Example Image](fig4.png)

Higher values of $X_{i}\beta $ indicate a greater likelihood of the event occurring. Care must be taken in interpreting estimation results in this context. A one-unit change in $X_{i}$ corresponds to a $\beta $ change in the z-score of Y. The estimated curve takes the form of an S-shaped cumulative normal distribution.

### ML estimation
Consier a simple regression with one independent variable $x_i$ for simplicity. To specify the liklihood equation, let $p_i=P(y_i=1|x_i)$ and $1-p_i=P(y=0|xi)$. Then, the likelihood is the product of $p_i$ for all observations for where $y=1$ multiplied by the product of $1-_i$ for which $y=0$: $$L(\beta_0, \beta_1|y_,x_i)=\prod\limits_{i=1}^{n1}p_{i}\prod\limits_{i=1}^{n2}1-p_{i},$$ where $n1$ is the number of observations where $y_{i}=1$ and $n2$ is the number of observations where $y_{i}=0$.

This can be expressed as substituting the $pi= P(y=1|X)=F(X\beta)$ $$L(\beta_0, \beta_1|y_,x_i)=\prod\limits_{i=1}^{n1}F(\beta_0+\beta_1x_i)\prod\limits_{i=1}^{n2}[1-F(\beta_0+\beta_1x_i )]$$. 

Taking the logarithm, we obtain the log-likelihood function $$\ln (L)=\sum\limits_{i=1}^{n1}\ln [F(\beta_0+\beta_1x_i )]\sum\limits_{i=1}^{n2}\ln [1-F(\beta_0+\beta_1x_i)]$$. Maximizing $\ln (L)$ with respect to $\beta $ gives the maximum likelihood (ML) estimator of $\beta $.

### Interpretation
For each observation of independent variables, we can estimate the predicted probabilities. These predicted probabilities can be summarized as a range, from the minimum to the maximum. The predicted probability for the $i$th individual is denoted as $\widehat{P}(y_{i}=1|X)=F(X\widehat{\beta })$. For positive $\beta$ values, $F(X\widehat{\beta })$ yields the minimum if we take the lower extreme of $x_{i}$. Conversely, if $\beta $ is negative, the lower extreme value of a variable will return the maximum probability. It's important to exercise caution with extreme values (outliers) as they can lead to misleading interpretations. In the presence of outliers, it may be more appropriate to consider the 5th and 95th quantiles instead of the minimum and maximum.

__Interpretation:Example(Probit)__
- $y_{i}=1$ if the ith individual travels by own vehicles, 0=otherwise
- $x_{i}=$ Difference in the travel time between public transport and own
vehicle( in minutes)
- $p_{i}=\Phi (X_{i}\beta )=\Phi (\alpha +\beta x_{i})=\Phi (Z)$
- $\widehat{p}_{i}=\Phi (\underset{(-0.161)}{-0.0644}+\underset{(2.95)}{0.0299}x_{i})$
- If $x_{i}=20:\Phi (0.54),$ hence $P(z\leq 0.54)=0.7054$
- If $x_{i}=5:\Phi (0.085),P(z\leq 0.085)=0.53387$

#### Interpretation:Marginal effects 


In bivariate linear regression, if the coefficient on $x_{1}$ is $\beta_{1}$, then a 1-unit increase in $x_{1}$ increases $y$ by $\beta_{1}$. In a probit model, a 1-unit increase in $x_{1}$ increases the z-score of $P(y=1)$ by $\beta_{1}$. An increase in $x_{1}$ has a constant effect on $y^{\ast }$, but this doesn't translate into a constant effect on the original $y$ or its probability. For a logit/probit model, $p=P(y=1|x_{1i})=F(\beta_0+\beta_1x_{1i} )$, where $F$ is either the cumulative distribution function $\Phi $ for a normal distribution or the cumulative distribution function $\Lambda $ for the logistic distribution. The marginal effect of $x_{k}$, the partial change in the probability due to a change in $x_{k}$, is given by$$$\frac{\partial p_{i}}{\partial x_{k}}=\frac{\partial F(X_{i}\beta )}{
\partial x_{k}}=\frac{dF(X\beta )}{dX\beta }\frac{\partial X\beta }{
\partial x_{k}}=f(X\beta )\beta _{k},$$ where $f$ is the probability density function. 

In a probit model, $\frac{\partial p{i}}{\partial x_{k}}=\phi (\beta_0+\beta_1x_{1i}+...+\beta_{ki} )\beta_{k}$. Since $f(X\beta )$ is always positive, the sign of $\widehat{\beta}_{k}$ indicates the direction of the relation between $x_{k}$ and $p_{i}$. The magnitude of the change in the probability given a change in $x_{k}$ is determined by the magnitude of $\widehat{\beta }_{k}$ and $f(X\beta )$. Consequently, $\frac{\partial p_{i}}{\partial x_{k}}$ varies with $x_{k}$ and values of other independent variables.

For example, $$\widehat{p}{i}=\Phi (\underset{(-0.161)}{-0.0644}+\underset{(2.95)}{0.0299}x{i})$$. $$\frac{\partial \widehat{p}{i}}{\partial x{i}}=(\widehat{\beta })\times \phi (\widehat{\alpha }+\widehat{\beta }x_{i})=(\widehat{\beta })\times \phi (z)=(0.0299)\times \underset{\text{Changes with }x_{i}}{\underbrace{\phi (-0.0644+0.0299x_{i})}}.$$ If $x_{i}=20$, $\frac{\partial \widehat{p}{i}}{\partial x{i}}=0.0299\times \phi (0.54)=0.0104$ where $\phi (0.54)=0.3448$. 

Since marginal effects depend on the level of all variables, we need to summarize both. There are two prominent methods. One method is to compute the average over all observations: Mean$\frac{\partial \widehat{p}_{i}}{\partial x_{ik}}=\frac{1}{N}\sum\limits_{i=1}^{N}f(X\beta )\beta_{k}$. Another method is to compute the marginal effect at the mean of the independent variables: $\frac{\partial \widehat{p}_{i}}{\partial x_{ik}}=f(\overline{X}\beta )\beta_{k}$. Both methods give identical results most of the time.

The marginal effect of a dummy variable is $\widehat{p}_{i}^{1}-\widehat{p}_{i}^{0}$. 

Exactly the same issues arise with the logit model as with the probit model (but the formulas differ). The predicted probability is $\widehat{P}(y_{i}=1|X)=F(X\widehat{\beta })$ where $F=\Lambda $. The marginal effects are $\frac{\partial \widehat{p}_{i}}{\partial x_{ik}}=\frac{\partial \Lambda (X\widehat{\beta })}{\partial x_{ik}}=\lambda (X\beta )\beta_{k}$, which is equal to $\frac{\exp (X\beta )}{[1+\exp (X\beta )]^{2}}\beta _{k}$. 
#### Logit in terms of odds: 
Logit was derived from a latent variable perspective so far. There is an alternative derivation based on the odds ratio. If some event occurs with probability $p=P(y=1|X)$, then the odds of it happening are $$O(p)=p/(1-p)$$. Odds indicate how often something happens relative to how often it doesn't happen: $$p=0\Rightarrow O(p)=0$$, $$p=0.25\Rightarrow O(p)=1/3$$ (odds are 1-to-3 against), $$p=0.75\Rightarrow O(p)=3/1$$ (odds are 3-to-1 in favor).

For original $y=(0,1)$, $y$ as a probability $(0$ to $1)$, and odds of $y=(0$ to $\infty )$, the log odds of $y=(-\infty $ to $\infty )$. $$p_{i}=P(y_{i}=1|X)=\frac{\exp (X\beta )}{1+\exp (X\beta )},\Rightarrow (0 \text{ to} 1)$$. $$1-p_{i}=1-P(y_{i}=1|X)=\frac{1}{1+\exp (X\beta )}.$$ In particular, the odds are $$\Omega (X)=\frac{p_{i}}{1-p_{i}}=\exp (X\beta )\Rightarrow (0 \text{ to }\infty ).$$ The log odds, known as logit, are $$\ln \Omega (X)=\ln \left( \frac{p_{i}}{1-p_{i}}\right) =X\beta \Rightarrow (-\infty \text{ to }\infty).$$  Since logit is the log of odds, logistic slope coefficients can be interpreted as the effect of a unit of change in the $X$ variable on the predicted logits with the other variables in the model held constant: $\ln \Omega (X)=X\beta $. $\frac{\partial \ln \Omega (X)}{\partial x_{k}}=\beta _{k}$. We can also drive the impact on a change in one variable on the odds ratio.

## An example ( taken from Gujarati)

The data used here is a random sample of 1,196 US males.1.
The variables used in the analysis are as follows:
_Smoker_ = 1 for smokers and 0 for nonsmokers
_Age_ = age in years
_Education_ = number of years of schooling
_Income_ = family income
_Pcigs_ = price of cigarettes in individual states in 1979
### Linear Probability model

The Model to be estimated:
$$y_i=\beta_0+ \beta_1 Age_i + \beta_2 Educ_i + \beta_3 Income_i+ \beta_4 Pcigs+u_i$$
where 
$y_i$=1 if the individual is smoker and 0 otherwise

In [24]:
## Please intsall these packages if not installed in your syestem
library(huxtable)
library(knitr)
library(tidyverse)

In [15]:
data <- read.csv("logit_smoke.csv", sep=",")
lpm <- lm(smoker~age+educ+income+pcigs79, data=data)
kable(tidy(lpm), digits=2, align='c', caption=
        "LPM model of to smoke or not to smoke.")




|    term     | estimate | std.error | statistic | p.value |
|:-----------:|:--------:|:---------:|:---------:|:-------:|
| (Intercept) |   1.12   |   0.19    |   5.96    |  0.00   |
|     age     |   0.00   |   0.00    |   -5.70   |  0.00   |
|    educ     |  -0.02   |   0.00    |   -4.47   |  0.00   |
|   income    |   0.00   |   0.00    |   0.63    |  0.53   |
|   pcigs79   |  -0.01   |   0.00    |   -1.80   |  0.07   |

Notice that all the variables, except income, are individually statistically significant
at least at the 10% level of significance.

The interpretation of the regression coefficients is as follows. If we hold all other variables constant, the probability of smoking decreases at the rate of $\approx$0.005 as a person ages, probably due to the adverse impact of smoking on health. Likewise, other things being equal, an increase
in schooling by one year decreases the probability of smoking by 0.02. Similarly, if
the price of cigarettes goes up by a dollar, the probability of smoking decreases by
 $\approx$ 0.005, holding all other variables constant.

In [16]:
summary(lpm)$r.squared

[1] 0.03876969

 The R2 value of  $\approx$ 0.038 seems very low,
but one should not attach much importance to this because the dependent variable is
nominal, taking only values of 1 and zero. The LPM  is not the preferred choice for modeling indicator variables for the reasons I mentioned in the class. Let us estimate it via Probit and Logit models. use **glm()** function to estimate the models.
### Logit model
The results of the logit model is given below:

In [17]:
s_logit<- glm(smoker ~ age + educ + income + pcigs79, family=binomial(link="logit"), 
                   data=data)
kable(tidy(s_logit), digits=4, align='c', caption=
        " 1. Logit model of to smoke or not to smoke")



|    term     | estimate | std.error | statistic | p.value |
|:-----------:|:--------:|:---------:|:---------:|:-------:|
| (Intercept) |  2.7451  |  0.8292   |  3.3105   | 0.0009  |
|     age     | -0.0209  |  0.0037   |  -5.5773  | 0.0000  |
|    educ     | -0.0910  |  0.0207   |  -4.4021  | 0.0000  |
|   income    |  0.0000  |  0.0000   |  0.6583   | 0.5104  |
|   pcigs79   | -0.0223  |  0.0125   |  -1.7895  | 0.0735  |

The interpretation of the logit model in  is as follows: each slope coefficient
shows how the log of the odds in favor of smoking changes as the value of the X
variable changes by a unit 
Let us examine these results. The variables age and education are highly statistically significant and have the expected signs. As age increases, the value of the logit
decreases, perhaps due to health concerns – that is, as people age, they are less likely
to smoke. Likewise, more educated people are less likely to smoke, perhaps due to
the ill effects of smoking. The price of cigarettes has the expected negative sign and
is significant at about the 7% level. Keeping all other variables constant, the higher the price of cigarettes,
the lower is the probability of smoking. Income has no statistically visible impact on
smoking, perhaps because expenditure on cigarettes may be a small proportion of
family income.
The interpretation of the various coefficients is as follows: holding other variables
constant, if, for example, education increases by one year, the average logit value goes
down by$\approx$0.09, that is, the log of odds in favor of smoking goes down by about 0.09.
Other coefficients are interpreted similarly.
#### Marginal effects
In the LPM the slope coefficient measures the marginal effect of a unit change in
the explanatory variable on the probability of smoking, holding other variables
constant. This is not the case with the logit model, for the marginal effect of a unit change in the explanatory variable not only depends on the coefficient of
that variable but also on the level of probability from which the change is measured. But the latter depends on the values of all the explanatory variables in the
model. You can use functions from *mfx* packge to get the marginal effects



In [20]:
 library("mfx")
mlogit <-logitmfx(smoker ~ age + educ + income + pcigs79,data=data)

In [22]:
mlogit

Call:
logitmfx(formula = smoker ~ age + educ + income + pcigs79, data = data)

Marginal Effects:
              dF/dx   Std. Err.       z     P>|z|    
age     -4.8903e-03  8.7359e-04 -5.5979 2.170e-08 ***
educ    -2.1334e-02  4.8365e-03 -4.4111 1.029e-05 ***
income   1.1069e-06  1.6814e-06  0.6583   0.51033    
pcigs79 -5.2340e-03  2.9242e-03 -1.7899   0.07347 .  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

The interpretation of these marginal effects is as follows: Keeping all other variables constant, if mean
age increases by a year, it reduces the probability of smoking approximately by 0.005.
Likewise, if mean education increases by a year, the probability of smoking goes
down by about 0.021, Keeping all other variables constant. A unit changes in mean income has no statistically significant impact on the rate of change in the probability of smoking. A unit
increase in the mean price of a pack of cigarettes reduces the probability of smoking
by about .005, which is significant at about the 7% level


#### The language of the odds ratio (OR)
Instead of expressing logits in terms of probabilities, we can express them in terms
of the odds ratios (OR). For our example, using Stata 12, we obtained the odds ratios
shown in Table below gives the odds ratios, their standard errors, and their z values (instead
of the t statistic).

In [25]:
options(scipen=999)
orlogit <-logitor(smoker ~ age + educ + income + pcigs79,data=data)
(orlogit)

Call:
logitor(formula = smoker ~ age + educ + income + pcigs79, data = data)

Odds Ratio:
           OddsRatio    Std. Err.       z         P>|z|    
age     0.9793627211 0.0036617871 -5.5773 0.00000002443 ***
educ    0.9130425607 0.0188687267 -4.4021 0.00001072081 ***
income  1.0000047202 0.0000071705  0.6583       0.51036    
pcigs79 0.9779283903 0.0121969923 -1.7895       0.07354 .  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

 These odds ratios are obtained by exponentiating the coefficients given in Table 1. For example,
the coefficient of age in Table 1 is –0.020853. Therefore, we compute $e^{–0.020853}$ =
0.9793627, which gives the value of the odds ratio.To interpret the odds ratios, keep in mind that an OR greater than 1 indicates an increased chance that an event (smoking) will occur rather than it not occurring. An odds ratio of less than 1 suggests a decreased chance of the event occurring versus
not occurring. An OR of 1 means that the chances of an event occurring or not
occurring are even. Another property of the odds ratio is that it is invariant with
respect to the ordering of the variables. Thus, ORa/ORb = 1/(ORb/ORa), where a and
b are two events. You will observe that the odds ratios for age and education
are less than one, suggesting that the probability of smoking decreases with respect
to these variables; not a surprising finding. More specifically, the odds of smoking
decrease by about 2% as one ages, ceteris paribus. Similarly, an additional year of
education reduces the odds of smoking by about 8.7%, ceteris paribus. The OR for income is not much different from 1. Th e OR for picgs79 also does not seem much
different from one

### Probit Model

As we have disussed in the class in  the probit model the error term has the normal distribution.Although the numerical values of the logit and probit coefficients are different, qualitatively the results are similar: the coefficients of age, education, and price of cigarettes are individually significant at least at the 10% level. The income coefficient,however, is not significant.

In [26]:
s_probit <- glm(smoker ~ age + educ + income + pcigs79, family=binomial(link="probit"), 
                   data=data)
kable(tidy(s_probit), digits=2, align='c', caption=
        "Probit model of to smoke or not to smoke")



|    term     | estimate | std.error | statistic | p.value |
|:-----------:|:--------:|:---------:|:---------:|:-------:|
| (Intercept) |   1.70   |   0.51    |   3.34    |  0.00   |
|     age     |  -0.01   |   0.00    |   -5.70   |  0.00   |
|    educ     |  -0.06   |   0.01    |   -4.46   |  0.00   |
|   income    |   0.00   |   0.00    |   0.62    |  0.54   |
|   pcigs79   |  -0.01   |   0.01    |   -1.79   |  0.07   |

#### Marginal effects
The interpretations are similar to logit model

In [28]:
mprobit <-probitmfx(smoker ~ age + educ + income + pcigs79,data=data)

mprobit

Call:
probitmfx(formula = smoker ~ age + educ + income + pcigs79, data = data)

Marginal Effects:
                dF/dx     Std. Err.       z         P>|z|    
age     -0.0049204438  0.0008628909 -5.7023 0.00000001182 ***
educ    -0.0213399178  0.0047859798 -4.4588 0.00000824044 ***
income   0.0000010324  0.0000016740  0.6168       0.53740    
pcigs79 -0.0052348582  0.0029164385 -1.7949       0.07266 .  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1